br-external/package/linux_firmware_next/linux_firmware_next.sh

```diff
@@ -0,0 +1,155 @@
+PATH_TO_LINUX_FIRMWARE_NEXT=$PATH_TO_LINUX_FIRMWARE_NEXT_PARENT"/"$LINUX_FIRMWARE_NEXT_FILENAME
+
+# Validate conditions to perform current script (if the conditions are not met, skip script execution):
+#    direcory with linux-firmare-next must exit, and be pointed at path $PATH_TO_LINUX_FIRMWARE_NEXT
+#    hash file (linux-firmware.hash) must exist and be located under the path $PATH_TO_HASH (required to calculate new hashes)
+#    post-image-efi-gpt.sh must exist and be located under the path $PATH_TO_POST_IMAGE_EFI_GPT (required to increase disk.img size)
+validate_condtitions()
+{
+    # Creates folders where linux-firmware-next.tar.xz will be placed
+    if [ ! -d $PATH_TO_DL ]; then
+        echo $PATH_TO_DL' does not exist.'
+        mkdir $PATH_TO_DL
+    fi
+
+    if [ ! -d $PATH_TO_DL_LINUX_FIRMWARE ]; then
+        echo $PATH_TO_DL_LINUX_FIRMWARE' does not exist.'
+        mkdir $PATH_TO_DL_LINUX_FIRMWARE
+    fi
+
+    if [ ! -d $PATH_TO_LINUX_FIRMWARE_NEXT ]; then
+        echo $PATH_TO_LINUX_FIRMWARE_NEXT' does not exist. Skip introduction linux-firmware-next.'
+        exit 0
+    else
+        echo $PATH_TO_DL_LINUX_FIRMWARE' exist. Firmware next can be copied to build folder'
+    fi
+
+    # Check if the linux-firmware.hash file exists, if not, skip current script execution
+    if [ ! -f $PATH_TO_HASH ]; then
+        echo "Error! File "$PATH_TO_HASH" does not exist. Skip introduction linux-firmware-next."
+        exit 0
+    fi
+
+    # Check if the post-image-efi-gpt.sh file exists, if not, skip current script execution
+    if [ ! -f $PATH_TO_POST_IMAGE_EFI_GPT ]; then
+        echo "Error! File "$PATH_TO_POST_IMAGE_EFI_GPT" does not exist. Skip introduction linux-firmware-next."
+        exit 0
+    fi
+}
+
+# Create .tar file (linux-firmware-next.tar.xz) based on directory with linux-firmware-next
+make_tar()
+{
+    if [ ! -f $PATH_TO_TAR ]; then
+        echo $PATH_TO_TAR' does not exist. Need to be created.'
+        echo "tar cJf "$PATH_TO_DL_LINUX_FIRMWARE/linux-firmware-next.tar.xz $PATH_TO_LINUX_FIRMWARE_NEXT "(this operation may take a while)"
+        tar -C $PATH_TO_LINUX_FIRMWARE_NEXT_PARENT  --transform="s/$LINUX_FIRMWARE_NEXT_FILENAME/$LINUX_FIRMWARE_NEXT_PROPER_FILENAME/" -cJf $PATH_TO_TAR $LINUX_FIRMWARE_NEXT_FILENAME --checkpoint=.100
+    fi
+}
+
+# Update makefile linux_firmware.mk to change version (on version with linux firmware next)
+update_makefile()
+{
+    line=$(grep 'LINUX_FIRMWARE_VERSION =' package/linux-firmware/linux-firmware.mk)
+    output="LINUX_FIRMWARE_VERSION = next"
+
+    sed -i "s|$line|$output|" package/linux-firmware/linux-firmware.mk
+}
+
+# Update hash file (linux-firmware.hash):
+#    arg $1: file based on which the hash is calculated
+#    arg $2: string in the hash file that should be replaced 
+update_hash_file()
+{
+    line=$(grep $2 $PATH_TO_HASH)
+    sha256="$(sha256sum $1 | cut -d' ' -f1)"
+    filename=$(basename $1)
+    echo "sha256 " $sha256 " " $filename
```

Comment:
Is it good idea to print sha of every file?

Reply:
I changed this and it only shows when the file is actually being overwritten. I think that if we make a change to a file, it is worth informing the user about it